File: 2024/day_2/python/day_two.py
def calcuate_safety(data):
    increasing = data[0] < data[1]
    for index in range(len(data) - 1):
        diff = data[index + 1] - data[index]
        if increasing and (diff < 1 or diff > 3):
            return False
        if not increasing and (diff > -1 or diff < -3):
            return False
    return True

def main():

    output = []
    count = 0

    with open('input.txt', 'r') as file:
        for line in file:
            output.append([int(v) for v in line.split()])

    for a in output:
        if calcuate_safety(a):
            count = count + 1

    print(count)


if __name__ == "__main__":
    main()

File: 2024/day_2/typescript/day_two.ts
import * as fs from 'fs'

export const calcuate_safety = (list: number[]) => {
  const increasing = list[0] < list[1]
  for(let index = 0; index < list.length -1; index++) {
    const diff = list[index + 1] - list[index];
    if (increasing && (diff < 1 || diff > 3)){
      return false
    }
    if (!increasing && (diff > -1 || diff < -3)){
      return false
    }
  };
  return true;
}


const decode_file = (data: string): number[][] => {
  const output: number[][] = [];
  data.split('\n').filter(
    val => val.trim() !== ''
  ).forEach((val) => {
    output.push(
      val.split(/\s+/).map(val => parseInt(val))
    );
  })
  return output
}

function main() {
  const data: string = fs.readFileSync('input.txt', 'utf8')

  const levels = decode_file(data);

  let count = 0;

  levels.forEach((v) => {
    if(calcuate_safety(v)){
      count++;
    }
  })

  console.log(count)
}

if (require.main === module) {
  main();
}

